### ISSUE TITLE
Support "OpenGL" fullscreen apps

### ISSUE THREAD

**Issue by choco** [koekeishiya/chunkwm#49]
This is a different mode from apps taking non-native fullscreen space (like the option available in VLC) and from native fullscreen. This is mostly used by games, that initialise an OpenGL context on the whole screen.
With kwm this almost never worked, we should probably detect when the screen enters this state and pause/ignore any notification.

The `NSWorkspaceActiveDisplayDidChangeNotification` seems to be thrown when this happens.

**Comment by koekeishiya** [comment]
I haven't really looked into how these applications behave at all. Furhter investigation is necessary before anything is done regarding this.

**Comment by koekeishiya** [comment]
@choco I just tested a game engine I was working on at some point, using opengl (glfw for windowing), and it works flawlessly with chunkwm on my end with instructions as specified here: http://www.glfw.org/docs/latest/window_guide.html#window_full_screen

**Comment by choco** [comment]
I haven't tried to see how chunkwm behaves in respect to kwm, but at least with kwm the main issues were:
1. sometimes a crash
2. when quitting the app, or exiting fullscreen mode kwm wouldn't notice the resolution change, for example game was using a resolution, kwm would use that resolution instead of standard one, forcing a restart
3. kwm becomes unresponsive, not sure what happens, seems something stalls execution, all events get queued and after something timed out all got executed together.
These issues were reproduced playing League of Legends, weren't triggered everytime, seems the amount of time the game run influenced the probability of these issues to arise.
That's why I think halting notifications of any kind would maybe be a good solution.

On an unrelated note, just started today reading the codebase to start helping on something and I'm really impressed, really great work 👍

**Comment by choco** [comment]
Finally reinstalled League of Legends yesterday, and the issue now is incredibly strange.
Basically the game becomes unplayable, the mouse cursor doesn't move correctly, seems like mouse movement is buffered and played back, I really don't know how to explain and something similar for the key presses.

**Comment by koekeishiya** [comment]
I believe most games rely on raw_input, and so a running event tap in the way khd and the ffm plugin works, is not gonna play well with them.

**Comment by choco** [comment]
khd doesn't seem to cause any issue, it's just chunkwm, not sure how.... even kwm didn't behave like this, but anyway that's why it would probably be better to have a way to stop notifications, or maybe just ingnore?!?, when in this state

**Comment by koekeishiya** [comment]
> khd doesn't seem to cause any issue, it's just chunkwm, not sure how

Hmm I really have no idea which notifications that may be. I suspect it would be the ffm plugin at least. I have run many games with chunkwm and the tiling plugin running, with no problms at all.

I usually have to stop khd as it can sometimes cause keys to get "stuck" from the games perspective. I don't use the ffm plugin at all anymore though.

**Comment by choco** [comment]
Yeah you're right, ffm seems to be the culprit here, no idea why the same issue didn't present itself with kwm; by a quick look the implementation seems to be the same.
I'll try to take a better look asap.
